docs/notes/2.23.x.md

```diff
@@ -57,29 +45,70 @@ plugin](https://github.com/pantsbuild/pants/tree/2.23.x/testprojects/pants-plugi
 for inspiration. To opt into the feature set the flag
 `--enable-target-origin-sources-blocks`.
 
-### General
+#### New call-by-name syntax for @rules
+
+Pants has a new mechanism for `@rule` invocation in backends. In this release the following backends were migrated to use this new mechanism. There should not be any user-visible effects, but please be on the lookout for any unusual bugs or error messages.
+
+- `cc`
+- `cue`
+- `debian`
+- `java.lint`
+- `kotlin.lint`
+- `makeself`
+- `scala.lint`
+- `semgrep`
+- `sql`
+- `swift`
+- `trufflehog`
+- `yamllint`
 
-New advanced options `--file-downloads-retry-delay` and `--file-downloads-max-attempts` allow configuration of the retry behavior when retryable errors occur while Pants is downloading files, for example, while downloading a `http_source` source.
+#### BUILD files
+
+Support for parametrizing grouped parametrizations. (e.g. `**parametrize(resolve=parametrize("a", "b"), ...)`). This works for `___defaults__` as well, as long as it is specified per target type rather than using `all`.
+
+### Remote caching/execution
+
+The deprecation for the `[GLOBAL].remote_auth_bearer_token_path` option has expired. Use [the `[GLOBAL].remote_auth_bearer_token = "@/path/to/file"` option](https://www.pantsbuild.org/2.23/reference/global-options#remote_oauth_bearer_token) instead.
 
 ### Goals
 
-#### Package
+#### `package`
+
+The `output_path` field present on targets which can be packaged by `pants package` (for instance [`pex_binary(output_path=...)`](https://www.pantsbuild.org/2.23/reference/targets/pex_binary#output_path)) is now based on a template so that you can use parts of `output_path`'s default behavior when overriding it on a target. For example, you can use the template replacement `${{spec_path_normalized}}` to obtain the default output directory for the target (i.e., the directory in which the target lives with slashes replaced by dots).
+
+#### `test`
 
-The `output_path` field present on targets which can be packaged by `pants package` is now based on a template so that you can use parts of `output_path`'s default behavior when overriding it on a target. For example, you can use the template replacement `${{spec_path_normalized}}` to obtain the default output directory for the target (i.e., the directory in which the target lives with slashes replaced by dots).
+A [new option `--experimental-report-test-result-info`](https://www.pantsbuild.org/2.23/reference/goals/test#experimental_report_test_result_info) is added to the `[test]` config section. Enabling this option will
+produce a file on disk with information that would tell you more about the test results. For now, it reports only
+the source of the results. The tests might have been executed locally or remotely, but they might have been retrieved
+from the local or remote cache, or be memoized.
```

Comment:
I know you are copy editing here: But as a tangent, what is the case where the result is memorized but *not* from a local or remote cache?

Reply:
Good question, it may be referring to pantsd in-memory caching. That is, the three layers of caching:

1. in-memory, within a single Pantsd session
2. local cache, on-disk within ~/.cache/pants/lmdb_store` (etc.)
3. remote cache